src/nri/sampler.go

```diff
@@ -37,7 +39,7 @@ type DockerClient interface {
 }
 
 // NewSampler returns a ContainerSampler instance.
-func NewSampler(fetcher raw.Fetcher, docker DockerClient, exitedContainerTTL time.Duration) (*ContainerSampler, error) {
+func NewSampler(fetcher raw.Fetcher, docker DockerClient, exitedContainerTTL time.Duration, config config.ArgumentList) (*ContainerSampler, error) {
```

Comment:
I don't like the idea of having such a high-level concept as the argument list passed this down to the code. Have you considered any alternatives, e.g. modeling this new functionality outside of the `Inspector`? Could perhaps be something like `Informer` with the `Info` method, and pass a nil one `if $FARGATE || cs.config.DisableStorageMetrics`

Reply:
I though to convert the argument struct to a config struct sanitized but there is nothing to sanitize , and i thought it was better to keep it simple. 
I could add optional parameters and add the bool parameter in the fetcher perhaps.
Let me have a look about the informer approach

Reply:
I reviewed the solution and all docker api calls are managed by DockerClient interface so it would be a bit confusing to call the info method of the api from other object i think.